miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultProjectService.java

```diff
@@ -229,14 +240,29 @@ private void applyChanges(Project original, Project project) {
   }
 
 
-  public void applySetChangesContacts(Set<ProjectContactsAndRole> to, Set<ProjectContactsAndRole> from) {
+  public void applySetChangesContacts(List<ProjectContact> to, List<ProjectContact> from) {
+    for (int i = 0; i < to.size(); i++) {
+      boolean remover = true;
+      for (int j = 0; j < from.size(); j++) {
+        if (to.get(i).getContact().getId() == from.get(j).getContact().getId()) {
+          to.get(i).setContactRole(from.get(j).getContactRole());
+          remover = false;
+          break;
+        }
+      }
+      if (remover) {
+        to.remove(i);
+      }
+    }
```

Comment:
This can all be replaced by checking the contact role ID in addition to the contact ID in the `removeIf...noneMatch...` below